### ISSUE TITLE
Select all action in multi-select

### ISSUE THREAD

**Issue by fontsie** [ngx-formly/ngx-formly#1516]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request
```

**Current behavior**

We have the feature request from a client to have also the possibility to have a "select all/none" option on the multi-select. Right now (as per standard material select ctrl) there's no such option foreseen. One would have to click the options one by one which on a large list is impossible.

**Expected behavior**

Our current idea is to have a dedicated option at the very top that allows to select all others. Is there already an implementation for this in ngx-formly or would this be an option to add it as a configuration to the current ngx-formly material select?

**Minimal reproduction of the problem with instructions**

The current multi-select:  https://stackblitz.com/edit/ngx-formly-ui-material-8izvgb

**Comment by juristr** [comment]
@aitboudad hey, @fontsie is a work mate of mine and we were discussing this today. Curious what you think about it or how we could implement something like this (we're happy to provide a PR in case) 😃

**Comment by aitboudad** [comment]
@juristr I think we should provide him a gift then :smiley: 

there is an issue in material which track a similar feature https://github.com/angular/material2/issues/9149,
My preferred implementation is adding a `select all` button on the top of select list like the following example: https://stackblitz.com/edit/angular-material-select-all-j6pcc1

PR welcomed!

**Comment by juristr** [comment]
> My preferred implementation is adding a select all button on the top of select list like the following example: https://stackblitz.com/edit/angular-material-select-all-j6pcc1

That's exactly what I intended 👍. Do you think this should be configurable or by default when having a multi-select option list?

**Comment by aitboudad** [comment]
> Do you think this should be configurable or by default when having a multi-select option list?

it should be configurable, since it may not be useful for a small list of options.

**Comment by juristr** [comment]
Yep, naming and cache invalidation, isn't it :wink:

Makes sense, didn't think about the label option which is handy for localization purposes as well 👍

**Comment by aitboudad** [comment]
To make it simple `selectAllOption` can be `null` or `string`.
I wanted this to be part of `5.1`, let me know if I can help!

**Comment by juristr** [comment]
@aitboudad should be simple to do. Gonna give it a look later today or at latest tomorrow. Will let u know

**Comment by juristr** [comment]
PR merged, will be released with `v5.1` //cc @aitboudad
